mod/admin.php

```diff
@@ -1587,8 +1594,8 @@ function admin_page_users(App $a)
 	/* get pending */
 	$pending = q("SELECT `register`.*, `contact`.`name`, `user`.`email`
 				 FROM `register`
-				 LEFT JOIN `contact` ON `register`.`uid` = `contact`.`uid`
-				 LEFT JOIN `user` ON `register`.`uid` = `user`.`uid`;");
+				 JOIN `contact` ON `register`.`uid` = `contact`.`uid`
```

Comment:
Why have you replaced the ```LEFT JOIN``` with ```JOIN```?

Reply:
Because in some case, the pending list can end up being populated with the entire `contact` table.

Reply:
In which cases could that be? ```LEFT JOIN``` in comparison to ```INNER JOIN``` allows for joins where the right side needn't to exist.

BTW: I would always prefer ```INNER JOIN``` instead of ```JOIN```.

Reply:
It happened to me yesterday while I was conducting registration tests, I refreshed the admin users page to approve the 5th or 6th account I created and the screen blew in my face with thousands of contact names in the pending list.

I don't think it is the normal case but if the contact and user related to a registration row don't exist, nothing *else* should turn up in the results of the pending users query.

Reply:
When the right side doesn't exist, the selected rows will be ```NULL``` - but there never will be more than one row. It can only be something different. I have to check this - especially if it does make sense here to use something different than an "inner join".

Reply:
Basically, we expect corresponding rows in `contact` and `user`, so the `LEFT JOIN` is unwarranted here even if it will work most of the time.

Reply:
I guess I know why the query had had problems in your case. I guess that it is expected to display the "self" contact - but "self" isn't queried.

I had a look at the git history. There wasn't any hint why it was realized as a "left join", so an "inner join" indeed was correct.

Only thing is: Please don't replace other "left join" - most of them are expected that way, I think :-)

Reply:
Ha ha, this isn't a crusade about `LEFT JOIN`, I changed this specific one because it gave me trouble. I wish I didn't have to spend time figuring out why my admin page was suddenly broken. Initially I thought that a bot registered thousands of accounts through the registration form temporary open for my tests, including harmful HTML that wouldn't have been properly escaped.